### ISSUE TITLE
[BUG] TypeError: Cannot read properties of undefined (reading 'serializeBinaryToWriter')

### ISSUE THREAD

**Issue by winter-olaf** [google/mediapipe#4525]
### Have I written custom code (as opposed to using a stock example script provided in MediaPipe)

Yes

### OS Platform and Distribution

macos

### Programming Language and version

TypeScript ^4.9.5
next: 12.3.4

### MediaPipe version

0.10.1

### Solution

FaceLandmarker

### Describe the actual behavior

```shell
1. After refresh browser, `SyntaxError: Unexpected token 'export'` occurred. Only using `import`
2. If using require instead import, it works! in development mode... but doesn't work in production mode(after next build).
```

### Describe the expected behaviour

```shell
@mediapipe/tasks-vision package imported after refresh and works well
```

### Standalone code/steps you may have used to try to get what you need

First of all, I used only `import`.
```shell

import { FaceLandmarker, FilesetResolver } from '@mediapipe/tasks-vision';

useEffect(() => {
      const filesetResolver = await FilesetResolver.forVisionTasks(
        'https://cdn.jsdelivr.net/npm/@mediapipe/tasks-vision@latest/wasm'
      );
      const faceLandmarker = await FaceLandmarker.createFromOptions(filesetResolver, {
        baseOptions: {
          modelAssetPath: `https://storage.googleapis.com/mediapipe-models/face_landmarker/face_landmarker/float16/1/face_landmarker.task`,
          delegate: 'GPU',
        },
        outputFaceBlendshapes: true,
        runningMode: mediaPipeRunningMode,
        numFaces: 1,
      });

      setFaceLandmarker(faceLandmarker);
  }
};
}, []);
```

Second, I added mediapipe package using `require`. 
Like this:

```typescript
  useEffect(() => {
    initFaceLandmarker();
  }, []);

  const initFaceLandmarker = async () => {
    try {
      const { FaceLandmarker, FilesetResolver } = require('@mediapipe/tasks-vision');

      const filesetResolver = await FilesetResolver.forVisionTasks(
        'https://cdn.jsdelivr.net/npm/@mediapipe/tasks-vision@latest/wasm'
      );
      const faceLandmarker = await FaceLandmarker.createFromOptions(filesetResolver, {
        baseOptions: {
          modelAssetPath: `https://storage.googleapis.com/mediapipe-models/face_landmarker/face_landmarker/float16/1/face_landmarker.task`,
          delegate: 'GPU',
        },
        outputFaceBlendshapes: true,
        runningMode: mediaPipeRunningMode,
        numFaces: 1,
      });

      setFaceLandmarker(faceLandmarker);
    } catch (e) {
      console.log(e);
    }
  };
```

In this way, faceLandmarker works after refresh. In development mode (next dev)

But, after build(next build) and start(next start), FaceLandmarker.createFromOptions occurred error:

```
TypeError: Cannot read properties of undefined (reading 'serializeBinaryToWriter')
    at proto.mediapipe.tasks.core.proto.BaseOptions.serializeBinaryToWriter (1632d6f7-920cbf6799cbd25b.js:1:424479)
    at jspb.BinaryWriter.writeMessage (1632d6f7-920cbf6799cbd25b.js:1:212366)
    at proto.mediapipe.tasks.vision.face_landmarker.proto.FaceLandmarkerGraphOptions.serializeBinaryToWriter (1632d6f7-920cbf6799cbd25b.js:1:533510)
    at jspb.BinaryWriter.writeMessage (1632d6f7-920cbf6799cbd25b.js:1:212366)
    at jspb.Message.serializeBinaryExtensions (1632d6f7-920cbf6799cbd25b.js:1:237869)
    at proto.mediapipe.CalculatorOptions.serializeBinaryToWriter (1632d6f7-920cbf6799cbd25b.js:1:255467)
    at jspb.BinaryWriter.writeMessage (1632d6f7-920cbf6799cbd25b.js:1:212366)
    at proto.mediapipe.CalculatorGraphConfig.Node.serializeBinaryToWriter (1632d6f7-920cbf6799cbd25b.js:1:320540)
    at jspb.BinaryWriter.writeRepeatedMessage (1632d6f7-920cbf6799cbd25b.js:1:219978)
    at proto.mediapipe.CalculatorGraphConfig.serializeBinaryToWriter (1632d6f7-920cbf6799cbd25b.js:1:314819)
```

I guess Next JS caused problem, so I tried using only `React`, not `Next`.

```jsx
import { useState, useEffect } from "react";
import reactLogo from "./assets/react.svg";
import viteLogo from "/vite.svg";
import "./App.css";
import { FaceLandmarker, FilesetResolver } from "@mediapipe/tasks-vision";

function App() {
  const [count, setCount] = useState(0);

  useEffect(() => {
    temp();
  }, []);

  const temp = async () => {
    try {
      const filesetResolver = await FilesetResolver.forVisionTasks(
        "https://cdn.jsdelivr.net/npm/@mediapipe/tasks-vision@latest/wasm"
      );
      console.log(filesetResolver);
      const faceLandmarker = await FaceLandmarker.createFromOptions(
        filesetResolver,
        {
          baseOptions: {
            modelAssetPath: `https://storage.googleapis.com/mediapipe-models/face_landmarker/face_landmarker/float16/1/face_landmarker.task`,
            delegate: "GPU",
          },
          outputFaceBlendshapes: true,
          runningMode: "VIDEO",
          numFaces: 1,
        }
      );
      console.log(faceLandmarker);
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank" rel="noreferrer">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank" rel="noreferrer">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <p>
          Edit <code>src/App.jsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </>
  );
}

export default App;
```

Then build, start(vite preview)...

![image](https://github.com/google/mediapipe/assets/40337311/6335f410-43ee-421d-80ca-9ec2e4346f39)

Then it works!
Is there a right way to use `@mediapipe/tasks-vision` in nextJS?

**Comment by kuaashish** [comment]
@winter-olaf,

Could you please confirm that you are still looking for resolution Or the issue has been resolved from your end? Thank you!

**Comment by winter-olaf** [comment]
@kuaashish,

Now I using this package in `https://oomf.chat/preview`.

1. I guessed above problem occured in `Next.js` - Server Side Rendering (React Server Component), so tried to use next/dynamic(To apply Client Side Rendering) but also not worked. (I don't know this is 100% **correct** way to client side rendering)
2. Tried to use this package in `React.js` Project - Client Side Rendering, It worked.
3. I cannot guess exact reason.
4. So I copy vision_bundle.js file to project's local path and finally, It works.
5. I didn't tried this in React 13's CSR. (But I guess that will be work...)

Now I stopped testing because my project is about to end. Thank you for replying.
